app/src/main/java/org/sagebionetworks/research/mpower/MpPhoneAuthActivity.java

```diff
@@ -0,0 +1,152 @@
+package org.sagebionetworks.research.mpower;
+
+import static org.sagebionetworks.research.mpower.research.MpIdentifier.AUTHENTICATE;
+
+import android.annotation.SuppressLint;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.annotation.VisibleForTesting;
+import android.widget.TextView;
+
+import com.google.common.base.Strings;
+
+import org.researchstack.backbone.ResearchStack;
+import org.sagebionetworks.bridge.android.manager.AuthenticationManager;
+import org.sagebionetworks.bridge.android.manager.dao.AccountDAO;
+import org.sagebionetworks.bridge.rest.model.Phone;
+import org.sagebionetworks.bridge.rest.model.UserSessionInfo;
+import org.sagebionetworks.research.mpower.researchstack.framework.MpDataProvider;
+import org.sagebionetworks.research.mpower.researchstack.framework.MpTaskFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+
+import dagger.android.support.DaggerAppCompatActivity;
+import rx.subscriptions.CompositeSubscription;
+
+public class MpPhoneAuthActivity extends DaggerAppCompatActivity {
```

Comment:
I moved the ViewModel to android-sdk. I think that's fine since the VM does most of the logic and the activity now does barely anything and most of it is mPower specific